app/assets/javascripts/actions/revisions_actions.js

```diff
@@ -9,9 +11,349 @@ import {
 import { fetchWikidataLabelsForRevisions } from './wikidata_actions';
 import logErrorMessage from '../utils/log_error_message';
 import request from '../utils/request';
+import moment from 'moment';
+import { stringify } from 'query-string';
+import { ORESSupportedWiki, PageAssessmentGrades, PageAssessmentSupportedWiki } from '../utils/article_finder_language_mappings';
+import { url } from '../utils/wiki_utils';
+import { chunk, flatten } from 'lodash-es';
+import { queryUrl } from '../utils/article_finder_utils';
+
+const fetchAll = async (API_URL, params, continue_str) => {
+  const allData = [];
+  let continueToken;
+  let hasMore = true;
+  while (hasMore) {
+    let response;
+    if (continueToken) {
+      params[continue_str] = continueToken[continue_str];
+      params.continue = continueToken.continue;
+    }
+    try {
+      response = await request(`${API_URL}?${stringify(params)}&origin=*`);
+      if (!response.ok) {
+        throw response;
+      }
+    } catch (e) {
+      return allData;
+    }
+    const json = await response.json();
+    allData.push(...json.query.usercontribs);
+    if (json.continue) {
+      continueToken = json.continue;
+    } else {
+      hasMore = false;
+    }
+  }
+  return allData;
+};
+const fetchAllRevisions = async (API_URL, days, usernames, wiki, course_start, last_date) => {
+  let ucend;
+  if (moment(last_date).subtract(days, 'days').isBefore(course_start)) {
```

Comment:
If we can avoid using `moment` here, I think that would be preferable. All the dates in `course` are UTC and that's also what the MediaWiki API uses. I'd like, eventually, to replace `moment` with a lighter library.

Reply:
hmm. I guess I could, although it wouldn't be as concise as this. Though I guess I could create a helper function to extract the manual date handling.